milesightct101/sensor.go

```diff
@@ -129,6 +136,15 @@ func (n *CT101) Reconfigure(ctx context.Context, deps resource.Dependencies, con
 		return err
 	}
 
+	// set the interval if one was provided
+	// we do not send a default in case the user has already set an interval they prefer
+	if cfg.Interval != nil && *cfg.Interval != 0 {
+		_, err = n.addIntervalToQueue(ctx, *nodeCfg.Interval, false)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
This is repeated, in a few places, please make it a function on cfg if that is a common config.

Reply:
`cfg` is specific to every model and `addIntervalToQueue` is implemented differently for every model. I could maybe to simplify this further by passing in `addIntervalToQueue` to a different function

Reply:
you could can pass the pointer in?

Reply:
we spoke offline, at a future date we can try moving `addIntervalToQueue` to `node`, that takes a struct that users fill in to handle endianness/units/size etc.

Reply:
we can also try using generics to remove the configs